### ISSUE TITLE
[Batch File] SET without /A breaks formatting

### ISSUE THREAD

**Issue by michaelblyons** [sublimehq/Packages#463]
Ordinary `SET` operations now clobber the rest of the line they are on.

#### Actual
```bat
  SET T=%TIME: =0%
::^^^ keyword.command.dosbatch
::    ^^^^^^^^^^^^ source.dosbatch
```
#### Expected
```bat
  SET T=%TIME: =0%
::^^^ keyword.command.dosbatch
::      ^^^^^^^^^^ variable.other.readwrite
```

**Comment by Briles** [comment]
This could be fixed by including the `variables` top-level context within `command_set` but it would be more complete to scope the additional constructs like so:

```Batch
  SET T=%TIME: =0%
::^^^ keyword.command.dosbatch
::    ^ variable.other.readwrite
::     ^ keyword.operator.assignment
::      ^^^^^^^^^^ variable.other.readwrite
```

**Comment by michaelblyons** [comment]
Would adding contexts also fix lines where `SET` is not the only thing on the line?

```bat
:: Contrived Example
IF "%FOO%" == "BAR" ( SET BAZ=42 )
::                  ^ punctuation.definition.group.begin
::                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.group
::                               ^ - punctuation.definition.group.end
```

The closing `)` is not matched and its group is never closed.

I agree on the additional scopes.

**Comment by wbond** [comment]
I believe something like https://github.com/sublimehq/Packages/blob/master/Batch%20File/Batch%20File.sublime-syntax#L41-L46 needs to be added to the direct child context of `command_set`.

**Comment by wbond** [comment]
This should be fixed by f50e344c9ae04364fcdd5e3adab77a32dcb8c641.

Feel free to suggest further improvements or open any issues you find. I did a little research before implementing what I did, but I am not a regular user of bat files.
